test/annotation_test.dart

```diff
@@ -221,12 +224,25 @@ void main() {
         expect(matched, isTrue);
       });
 
-      test('class annotated with type defined via package uri', () {
-        ClassElement annotatedClass =
-            _getAnnotationForClass(libElement, 'AnnotatedWithJson');
-        var annotation = annotatedClass.metadata.single;
-        var matched = matchAnnotation(JsonSerializable, annotation);
-        expect(matched, isTrue);
+      group('class annotated with type defined via package uri', () {
+        test('When running on a normal file system', () {
+          ClassElement annotatedClass =
+              _getAnnotationForClass(libElement, 'AnnotatedWithJson');
+          var annotation = annotatedClass.metadata.single;
+          var matched = matchAnnotation(JsonSerializable, annotation);
+          expect(matched, isTrue);
+        });
+
+        test('When running in barback', () {
+          var annotation = _mockElementAnnotation(
```

Comment:
Is there anyway to do this without mock silly? I worry about this being brittle...